src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/SettingsPropertyValue.cs

```diff
@@ -192,12 +204,20 @@ private static object GetObjectFromString(Type type, SettingsSerializeAs seriali
             // Convert based on the serialized type
             switch (serializeAs)
             {
+#pragma warning disable CS0618 // Type or member is obsolete
                 case SettingsSerializeAs.Binary:
-                    byte[] buffer = Convert.FromBase64String(serializedValue);
-                    using (MemoryStream ms = new MemoryStream(buffer))
+#pragma warning restore CS0618 // Type or member is obsolete
+                    if (SettingsProperty.EnableUnsafeBinaryFormatterInPropertyValueSerialization)
+                    {
+                        byte[] buffer = Convert.FromBase64String(serializedValue);
+                        using (MemoryStream ms = new MemoryStream(buffer))
+                        {
+                            return (new BinaryFormatter()).Deserialize(ms);
+                        }
+                    }
+                    else
                     {
-                        // Issue https://github.com/dotnet/runtime/issues/39295 tracks finding an alternative to BinaryFormatter
```

Comment:
>. [](start = 47, length = 1)

Technically the issue is not really closed, so we don't really have an alternative, we will now only throw if they try to use it, right?